src/main/java/org/csanchez/jenkins/plugins/kubernetes/StandardPlannedNodeBuilder.java

```diff
@@ -25,10 +26,11 @@
                     .cloud(cloud)
                     .build();
             displayName = agent.getDisplayName();
-            f = Futures.immediateFuture(agent);
-        } catch (IOException | Descriptor.FormException e) {
+            f = CompletableFuture.completedFuture(agent);
+        } catch (IOException | Descriptor.FormException | NoSuchAlgorithmException e) {
             displayName = null;
-            f = Futures.immediateFailedFuture(e);
+            f = new CompletableFuture();
+            ((CompletableFuture<?>) f).completeExceptionally(e);
```

Comment:
f can be initialized as a CompletableFuture so you avoid this cast.

Reply:
fixed in ff1d11f